config/eks/prometheus/config-appmesh.yaml

```diff
@@ -0,0 +1,99 @@
+receivers:
+  prometheus:
+    config:
+      global:
+        scrape_interval: 1m
+        scrape_timeout: 10s
+      scrape_configs:
+        - job_name: 'kubernetes-pod-appmesh-envoy'
+          sample_limit: 10000
+          metrics_path: /stats/prometheus
+          kubernetes_sd_configs:
+            - role: pod
+          relabel_configs:
+            - source_labels: [ __meta_kubernetes_pod_container_name ]
+              action: keep
+              regex: '^envoy$'
+            - source_labels: [ __address__, __meta_kubernetes_pod_annotation_prometheus_io_port ]
+              action: replace
+              regex: ([^:]+)(?::\d+)?;(\d+)
+              replacement: ${1}:9901
```

Comment:
does single $ works for you in regex here? I remember there was a known issue for OTel Prom Receiver which accepts double `$$`. See issue - https://github.com/open-telemetry/opentelemetry-collector/issues/1077#issuecomment-641067426